### PR TITLE
HashWheelTimerTest cleanup

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
HashWheelTimerTest has busy/wait and sleep statements which are not necessary. We also depend upon a com.google.common.base.Supplier which isn't necessary.

Modifications:
- Remove buys wait loops and timeouts where possible

Result:
HashWheelTimerTest more explicit in verifying conditions and less reliant on wait times.

### CODE REVIEW

File: common/src/test/java/io/netty/util/HashedWheelTimerTest.java [hunk 1]
```diff
@@ -185,54 +179,53 @@ public void testRejectedExecutionExceptionWhenTooManyTimeoutsAreAddedBackToBack(
...
             TimeUnit.MILLISECONDS, 32, true, 2);
-        timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
-        Timeout timeoutToCancel = timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
-        timeoutToCancel.cancel();
-        waitUntilNewTimeoutCanBeAdded(200, timer, 1000);
+        timer.newTimeout(createNoOpTimerTask(), 5, TimeUnit.SECONDS);
+        Timeout timeoutToCancel = timer.newTimeout(createNoOpTimerTask(), 5, TimeUnit.SECONDS);
+        assertTrue(timeoutToCancel.cancel());
+
+        Thread.sleep(tickDurationMs * 5);
```
Comment (netkins) [inline common/src/test/java/io/netty/util/HashedWheelTimerTest.java]: CRITICAL Remove this use of "Thread.sleep()". [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)

Comment (normanmaurer) [inline common/src/test/java/io/netty/util/HashedWheelTimerTest.java]: ignore this..

**Comment by normanmaurer** [comment]
@Scottmitch ship it

**Comment by normanmaurer** [comment]
@Scottmitch actually I think this should go into 4.0 as well if I am not wrong

**Comment by Scottmitch** [comment]
4.1 (00fc239) 4.0 (d941d54)

Yip this also belongs in 4.0 and merges cleanly after https://github.com/netty/netty/pull/5852#issuecomment-260507570

**Pull request closed without merging**